src/transformers/trainer.py

```diff
@@ -3831,25 +3831,45 @@ def evaluation_loop(
             if is_torch_xla_available():
                 xm.mark_step()
 
+            use_gather_object_in_gather_function = (
+                "use_gather_object" in inspect.signature(self.gather_function).parameters.keys()
+            )
```

Comment:
I think it will be better if you also check the version of accelerate when `use_gather_object=True`, so that we can ask the user to update their version of accelerate . Also, in case we do have the required version of accelerate, we can either use functools.partial or kwargs to in order to not have multiple if/else statement afterwards.

Reply:
@SunMarc I think it can be seperated as a two part.
> I think it will be better if you also check the version of accelerate when use_gather_object=True, so that we can ask the user to update their version of accelerate .

Which specific version should I use?

> Also, in case we do have the required version of accelerate, we can either use functools.partial or kwargs to in order to not have multiple if/else statement afterwards.

I would also love to use kwargs and functools. Unfortunately in the CI test there is a function which is just using
https://github.com/huggingface/transformers/blob/12b1620e615592fbf099d4ec44af7b9f2d1b48aa/tests/extended/test_trainer_ext.py#L67
https://github.com/huggingface/transformers/blob/12b1620e615592fbf099d4ec44af7b9f2d1b48aa/src/transformers/trainer_seq2seq.py#L178
It uses pure gather function

Reply:
WDYT about the revised version except for accelerate version test?

Reply:
Oh makes sense ! Thanks for the new version, it is much better. As for the accelerate version, it is available since v0.30.0 !

Reply:
@SunMarc Done!